var/spack/repos/builtin/packages/amdblis/package.py

```diff
@@ -22,17 +22,50 @@ class Amdblis(BlisBase):
 
     maintainers = ['amd-toolchain-support']
 
+    version('3.1', sha256='2891948925b9db99eec02a1917d9887a7bee9ad2afc5421c9ba58602a620f2bf')
+    version('3.0.1', sha256='dff643e6ef946846e91e8f81b75ff8fe21f1f2d227599aecd654d184d9beff3e')
     version('3.0', sha256='ac848c040cd6c3550fe49148dbdf109216cad72d3235763ee7ee8134e1528517')
     version('2.2', sha256='e1feb60ac919cf6d233c43c424f6a8a11eab2c62c2c6e3f2652c15ee9063c0c9')
 
+    variant(
+        'ilp64',
+        default=False,
+        description='Build with ILP64 support')
+
+    conflicts(
+        '+ilp64',
+        when='@:3.0.0',
+        msg='ilp64 is supported from amdblis 3.0.1 version onwards')
+
     def configure_args(self):
         spec = self.spec
         args = super(Amdblis, self).configure_args()
 
-        if spec.satisfies('@3.0 %aocc'):
-            """ To enabled Fortran to C calling convention for
-            complex types when compiling with aocc flang"""
+        if spec.satisfies('+ilp64'):
+            args.append('--blas-int-size=64')
+
+        """ To enabled Fortran to C calling convention for
+        complex types when compiling with aocc flang"""
+        if "@3.0 %aocc" in self.spec:
```

Comment:
We generally recommend using `satisfies()` for spec checks that are more than whether a variant is enabled:

```suggestion
        if self.spec.satisfies("@3.0 %aocc"):
```

Reply:
Incorporated review comments